### DISCUSSION TITLE
could not allocate 16.00 B without exceeding limit

### DISCUSSION THREAD

**Discussion by jieshier** [apache/doris#23646] [RFC]
doris集群版本：1.0.5
集群概述：2fe、4be；be服务器8核、32G
【查询报错】：
Error querying database. Cause: java.sql.SQLException: errCode = 2, detailMessage = Could not allocate aggregate expression input value ExecNode:Exprs:AGGREGATION_NODE (id=1):3011085ea56f4925-b405e73916911af0 could not allocate 16.00 B without exceeding limit. Error occurred on backend [10.0.12.6](http://10.0.12.6/) by fragment 3011085ea56f4925-b405e73 ### The error may exist in class path resource
【相关日志】：
Error occurred on backend 10.0.12.6 by fragment 253a2f3ad2654edc-94157bc3c1fd8a45
Memory left in process limit: 24.60 GB
 precise_code:1
W0830 10:45:59.945168 29401 mem_tracker.cpp:523] Could not allocate aggregate expression input value
ExecNode:Exprs:AGGREGATION_NODE (id=3):253a2f3ad2654edc-94157bc3c1fd8a3e could not allocate 16.00 B without exceeding limit.
Error occurred on backend 10.0.12.6 by fragment 253a2f3ad2654edc-94157bc3c1fd8a45
Memory left in process limit: 24.60 GB
Process: Limit=24.60 GB Total=3.93 MB Peak=10.86 GB
  RuntimeState:query:e5fa3d8900af47bd-a3645368e1a45c85: Limit=2.00 GB Total=2.11 MB Peak=2.11 MB
    RuntimeState:instance::e5fa3d8900af47bd-a3645368e1a45c85: Total=2.11 MB Peak=2.11 MB
    BufferedBlockMgr2:e5fa3d8900af47bd-a3645368e1a45c85: Limit=2.00 GB Total=0 Peak=0
  PlanFragmentExecutor:e5fa3d8900af47bd-a3645368e1a45c85:e5fa3d8900af47bd-a3645368e1a45c86: Limit=2.00 GB Total=0 Peak=0
  RuntimeState:query:c30319431d4e4515-83e8929a0e4b7208: Limit=2.00 GB Total=1.11 MB Peak=1.11 MB
    RuntimeState:instance::c30319431d4e4515-83e8929a0e4b7208: Total=1.11 MB Peak=1.11 MB
    BufferedBlockMgr2:c30319431d4e4515-83e8929a0e4b7208: Limit=2.00 GB Total=0 Peak=0
  PlanFragmentExecutor:c30319431d4e4515-83e8929a0e4b7208:c30319431d4e4515-83e8929a0e4b7209: Limit=2.00 GB Total=0 Peak=0
  RuntimeState:query:545bd2b7ce4947dc-a9287def44db2f16: Limit=2.00 GB Total=692.00 KB Peak=692.00 KB
    RuntimeState:instance::545bd2b7ce4947dc-a9287def44db2f16: Total=692.00 KB Peak=692.00 KB
    BufferedBlockMgr2:545bd2b7ce4947dc-a9287def44db2f16: Limit=2.00 GB Total=0 Peak=0
  PlanFragmentExecutor:545bd2b7ce4947dc-a9287def44db2f16:545bd2b7ce4947dc-a9287def44db2f17: Limit=2.00 GB Total=0 Peak=0
  RuntimeState:query:ca98e112289f451f-a9f7289393a27193: Limit=8.00 GB Total=8.00 KB Peak=8.00 KB
    RuntimeState:instance::ca98e112289f451f-a9f7289393a27193: Total=8.00 KB Peak=8.00 KB
    BufferedBlockMgr2:ca98e112289f451f-a9f7289393a27193: Limit=8.00 GB Total=0 Peak=0
  PlanFragmentExecutor:ca98e112289f451f-a9f7289393a27193:ca98e112289f451f-a9f7289393a27196: Limit=8.00 GB Total=0 Peak=0
  RuntimeState:query:ca98e112289f451f-a9f7289393a27193: Limit=8.00 GB Total=8.16 KB Peak=8.16 KB
    RuntimeState:instance::ca98e112289f451f-a9f7289393a27193: Total=8.16 KB Peak=8.16 KB
    BufferedBlockMgr2:ca98e112289f451f-a9f7289393a27193: Limit=8.00 GB Total=0 Peak=0
  PlanFragmentExecutor:ca98e112289f451f-a9f7289393a27193:ca98e112289f451f-a9f7289393a27194: Limit=8.00 GB Total=12.00 KB Peak=1.16 MB
    OlapScanner:ca98e112289f451f-a9f7289393a27194: Limit=8.00 GB Total=12.00 KB Peak=156.00 KB
  RuntimeState:query:253a2f3ad2654edc-94157bc3c1fd8a3e: Limit=8.00 GB Total=0 Peak=0
    RuntimeState:instance::253a2f3ad2654edc-94157bc3c1fd8a3e: Total=0 Peak=0
    BufferedBlockMgr2:253a2f3ad2654edc-94157bc3c1fd8a3e: Limit=8.00 GB Total=0 Peak=0
  PlanFragmentExecutor:253a2f3ad2654edc-94157bc3c1fd8a3e:253a2f3ad2654edc-94157bc3c1fd8a45: Limit=8.00 GB Total=0 Peak=0
W0830 10:45:59.945295 29401 internal_service.cpp:115] exec plan fragment failed, errmsg=Could not allocate aggregate expression input value
ExecNode:Exprs:AGGREGATION_NODE (id=3):253a2f3ad2654edc-94157bc3c1fd8a3e could not allocate 16.00 B without exceeding limit

**Comment by jieshier** [comment]
服务器监控以及doris集群监控各方面都很正常
上述问题发生在大表的联表查询中，我尝试过调整内存，但是依然报错，麻烦您这边可以帮我分析下可以做哪些优化吗。谢谢🙏

**Comment by LemonLiTree** [comment]
1.0.5？

**Comment by LemonLiTree** [comment]
When joining, put the large table on the left

**Reply by jieshier** [reply]
Yes,1.0.5。 both are large tables

**Reply by LemonLiTree** [reply]
<img width="578" alt="image" src="https://github.com/apache/doris/assets/109049295/4163ffa2-974d-4073-8960-e7dedfbd95bb">
It seems there is no such version
